kolibri/plugins/management/assets/src/actions.js

```diff
@@ -69,110 +79,142 @@ function _managePageTitle(title) {
  */
 
 /**
- * Do a POST to create new user
- * @param {object} payload
- * @param {string} role
+ * Does a POST request to assign a user role (only used in this file)
+ * @param {object} user
+ * Needed: id, facility, kind
  */
-function createUser(store, payload, role) {
-  const FacilityUserModel = FacilityUserResource.createModel(payload);
-  const newUserPromise = FacilityUserModel.save(payload);
-  // returns a promise so the result can be used by the caller
-  return newUserPromise.then((model) => {
-    // assign role to this new user if the role is not learner
-    if (role === 'learner' || !role) {
-      store.dispatch('ADD_USER', _userState(model));
-    } else {
-      const rolePayload = {
-        user: model.id,
-        collection: model.facility,
-        kind: role,
+function assignUserRole(user, kind){
+  const rolePayload = {
+        user: user.id,
+        collection: user.facility,
+        kind: kind,
       };
-      const RoleModel = RoleResource.createModel(rolePayload);
-      const newRolePromise = RoleModel.save(rolePayload);
-      newRolePromise.then((results) => {
-        FacilityUserModel.fetch({}, true).then(updatedModel => {
-          store.dispatch('ADD_USER', _userState(updatedModel));
-        });
-      }).catch(error => { coreActions.handleApiError(store, error); });
-    }
-  }).catch((error) => Promise.reject(error));
+    
+  return new Promise((resolve, reject) => {
+    RoleResource.createModel(rolePayload).save().then((roleModel)=>{
+      // add role to user's attribute here to limit API call
+      user.roles.push(roleModel);
+      resolve(user);
+
+    },(error) => reject(error));
+  });
+}
+
+/**
+ * Do a POST to create new user
+ * @param {object} stateUserData
+ *  Needed: username, full_name, facility, role, password
+ */
+function createUser(store, stateUserData) {
+  const userData = {
+    facility: stateUserData.facility,
+    username: stateUserData.username,
+    full_name: stateUserData.full_name,
+    password: stateUserData.password,
+  }
+
+  return new Promise((resolve, reject) => {
+    FacilityUserResource.createModel(userData).save().then((userModel) => {
+      
+      // only runs if there's a role to be assigned
+      if (stateUserData.kind != UserKinds.LEARNER) {
+
+        assignUserRole(userModel, stateUserData.kind).then(
+          userWithRole => resolve(userWithRole), 
+          error => reject(error)
+        );
+      
+      // no role to assign
+      }else resolve(userModel);
+
+    },(error) => reject(error));
```

Comment:
the linter will probably have some suggestions.

A general rule-of-thumb that the linter doesn't enforce is: opening and closing brackets, parens, etc should 'line up' with the same level of indentation.

E.g.

```javascript
  }).then(
    // dispatch newly created user
    newUser => store.dispatch('ADD_USER', _stateUser(newUser)),
    // send back error if necessary
    error => error
  );
```

not 

```javascript
  }).then(
      // dispatch newly created user
      newUser => store.dispatch('ADD_USER', _stateUser(newUser)),
      // send back error if necessary
      error => error
    );
```

because the opening `.then(` and closing `);` both should have 2 spaces.

For an example relevant here, see: https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/actions.js#L85